File: app/src/main/java/com/example/alarm/model/AppVisibilityTracker.java
package com.example.alarm.model;

import android.content.Context;
import android.os.PowerManager;

public class AppVisibilityTracker {
    private static PowerManager powerManager;

    public static void initialize(Context context) {
        powerManager = (PowerManager) context.getSystemService(Context.POWER_SERVICE);
    }

    public static boolean isScreenOn() {
        return powerManager.isInteractive();
    }
}

File: app/src/main/java/com/example/alarm/model/AppVisibilityTracker.kt
package com.example.alarm.model

import android.content.Context
import android.os.PowerManager

object AppVisibilityTracker {
    private lateinit var powerManager: PowerManager

    fun initialize(context: Context) {
        powerManager = context.getSystemService(Context.POWER_SERVICE) as PowerManager
    }

    fun isScreenOn(): Boolean {
        return powerManager.isInteractive
    }
}